internal/ethapi/api.coreth.go

```diff
@@ -0,0 +1,118 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package ethapi
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ava-labs/coreth/params"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+const (
+	nAVAX = 1_000_000_000
+
+	minBaseFee       = params.EtnaMinBaseFee // 1 nAVAX
+	maxNormalBaseFee = 100 * nAVAX
+
+	minGasTip       = 1 // 1 wei
+	maxNormalGasTip = 20 * nAVAX
+
+	slowFeeNumerator = 19 // 19/20 = 0.95
+	fastFeeNumerator = 21 // 21/20 = 1.05
+	feeDenominator   = 20
+)
+
+var (
+	bigMinBaseFee       = big.NewInt(minBaseFee)
+	bigMaxNormalBaseFee = big.NewInt(maxNormalBaseFee)
+
+	bigMinGasTip       = big.NewInt(minGasTip)
+	bigMaxNormalGasTip = big.NewInt(maxNormalGasTip)
+
+	bigSlowFeeNumerator = big.NewInt(slowFeeNumerator)
+	bigFastFeeNumerator = big.NewInt(fastFeeNumerator)
+	bigFeeDenominator   = big.NewInt(feeDenominator)
+)
+
+type Price struct {
+	GasTip *hexutil.Big `json:"maxPriorityFeePerGas"`
+	GasFee *hexutil.Big `json:"maxFeePerGas"`
+}
+
+type PriceOptions struct {
+	Slow   *Price `json:"slow"`
+	Normal *Price `json:"normal"`
+	Fast   *Price `json:"fast"`
+}
+
+// SuggestPriceOptions returns suggestions for what to display to a user for
+// current transaction fees.
+func (s *EthereumAPI) SuggestPriceOptions(ctx context.Context) (*PriceOptions, error) {
+	baseFee, err := s.b.EstimateBaseFee(ctx)
+	if err != nil || baseFee == nil {
+		return nil, err
```

Comment:
I think it's safe to use error wrapping especially for new code:

```suggestion
		return nil,fmt.Errorf("failed to estimate base fee: %w", err)
```

Reply:
Ah interesting; I think we should split this in two then. That if block did confuse me as well [previously](https://github.com/ava-labs/coreth/pull/789/files#r1952861528), so splitting it would be good for readability as well.

```go
if err != nil {
	return nil, fmt.Errorf("failed to estimate base fee: %w", err)
} else if baseFee == nil {
	return nil, nil
}
```

And let's error wrap as much as we can (within reasons, i.e. never in core/vm 😄) - nobody likes an error with no idea where it's coming from and what code path it took.